android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt

```diff
@@ -68,24 +66,47 @@ class PianoAnalyticsPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
             StandardMethodCodec(Codec())
         )
         channel.setMethodCallHandler(this)
-
-        consentsPlugin
-        consentsPlugin.onAttachedToEngine(flutterPluginBinding)
     }
 
     override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
         channel.setMethodCallHandler(null)
-        consentsPlugin.onAttachedToEngine(binding)
     }
 
     override fun onMethodCall(call: MethodCall, result: Result) {
         try {
+            var skipResult = false
```

Comment:
```suggestion
            val skipResult = call.method == "getVisitorId"
```

Reply:
skipResult is also used for getUser

Reply:
`val skipResult = call.method in METHODS_WITH_CUSTOM_RESULT`, but it's better to make something like:
```
val methodResult = when (call.method) { ... }
result.success(methodResult)
```
+
```
    private fun handleGetUser(): Map<String?, Any>? {
        return PianoAnalytics.getInstance().userStorage.currentUser?.let {
            mapOf(
                    "id" to it.id,
                    "category" to it.category
            )
        }
    }
```
+
```
private fun handleInit(call: MethodCall): Any? {
...
    return null
}